File: retroravelry/src/main/java/com/caseykulm/retroravelry/models/pack/FullPack.java
package com.caseykulm.retroravelry.models.pack;

import com.google.auto.value.AutoValue;

@AutoValue public abstract class FullPack {

  public abstract String color_family_id();
  public abstract String colorway();
  public abstract String dye_lot();
  public abstract int id();
  public abstract String personal_name();
  public abstract int project_id();
  public abstract int shop_id();
  public abstract String shop_name();
  public abstract String skeins();
  public abstract int stash_id();
  public abstract int yarn_id();
  public abstract String yarn_name();

}

File: retroravelry/src/main/java/com/caseykulm/retroravelry/models/pack/FullPack.kt
package com.caseykulm.retroravelry.models.pack

import com.squareup.moshi.Json

data class FullPack(
        @Json(name="color_family_id") val colorFamilyId: String?,
        val colorway: String?,
        val dye_lot: String?,
        val id: Int,
        val personal_name: String?,
        val project_id: Int,
        val shop_id: Int,
        val shop_name: String?,
        val skeins: String?,
        val stash_id: Int,
        val yarn_id: Int,
        val yarn_name: String?
)
